app/src/main/java/com/android/sample/ui/searchIngredient/SearchIngredientScreen.kt

```diff
@@ -188,7 +188,14 @@ private fun SearchDisplay(
                 Modifier.weight(ICON_SCANNER_WEIGHT)
                     .size(ICON_SCANNER_SIZE.dp)
                     .testTag(SCANNER_ICON)
-                    .clickable { navigationActions.navigateTo(Screen.CAMERA_SCAN_CODE_BAR) },
+                    .clickable {
+                      if (navigationActions.currentRoute() == Screen.FRIDGE_SEARCH_ITEM) {
+                        navigationActions.navigateTo(Screen.FRIDGE_SCAN_CODE_BAR)
+                      } else if (navigationActions.currentRoute() ==
+                          Screen.CREATE_RECIPE_SEARCH_INGREDIENTS) {
+                        navigationActions.navigateTo(Screen.CAMERA_SCAN_CODE_BAR)
+                      }
+                    },
```

Comment:
I feel as if this should be a parameter of the function, instead of checking the current route. Like you did for the popupmessage